WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -304,9 +313,52 @@ class PostCoordinator: NSObject {
         }
     }
 
+    /// Safely pauses sync for the post. If there are any outstanding operations
+    /// that can't be canceled, allowing them to finish. When the method returns,
+    /// it's guaranteed that no requests will be sent until resumed.
+    @MainActor
+    func pauseSyncing(for post: AbstractPost) async {
```

Comment:
I'm open to another ideas on how to implement it, but seemed like the least worst option.

It would've probably been easier to ask the coordinator to sync the rest of the changes, but I'm not sure it's feasible to handle the errors here: what if one of the images from the previous revisions keeps failing or what if there is a conflict? When publishing, we no longer _need_ to get these previous revisions synced – we have everything that's needed to publish.